project-koku/koku
# COST-2678 - create source and load test customer data for OCI

Fixes [COST-2678](https://issues.redhat.com/browse/COST-2678)

Changes proposed in this PR:
* add oci to test customer yaml
* add OCI static data yml file
* add OCI cost model
* load OCI data into koku


Testing Instructions:
1. clone this pr with this command `gh pr checkout 3691` and start the koku
2. add a customer and create sources with this command `make create-test-customer`
3. load oci data into koku with `make load-test-customer-data test_source=oci`
4. hit this endpoint `GET http://127.0.0.1:8000/api/cost-management/v1/reports/oci/costs/`
5. Expect to see 200 in the logs,
   ```    
         koku_server       | [2022-06-10 18:56:27,872] INFO None {'method': 'GET', 'path': '/api/cost-management/v1/reports/oci/costs/', 'status': 200, 'request_id': None, 'account': '10001', 'username': 'user_dev', 'is_admin': 'True', 'response_time': '60.20 ms'}
         koku_server       | [2022-06-10 18:56:27,874] INFO None "GET /api/cost-management/v1/reports/oci/costs/ HTTP/1.1" 200 
    6917
   ```

and  non-zero values under cost in the response. for example

```
{
    "meta": {
        "count": 10,
        "check_tags": false,
        "filter": {
            "time_scope_value": "-10",
            "time_scope_units": "day",
            "resolution": "daily"
        },
        "group_by": {},
        "order_by": {},
        "total": {
            "infrastructure": {
                "raw": {
                    "value": 7108.8,
                    "units": "USD"
                },
                "markup": {
                    "value": 1421.76,
                    "units": "USD"
                },
                "usage": {
                    "value": 0.0,
                    "units": "USD"
                },
                "total": {
                    "value": 8530.56,
                    "units": "USD"
                }
            },
            "supplementary": {
                "raw": {
                    "value": 0.0,
                    "units": "USD"
                },
                "markup": {
                    "value": 0.0,
                    "units": "USD"
                },
                "usage": {
                    "value": 0.0,
                    "units": "USD"
                },
                "total": {
                    "value": 0.0,
                    "units": "USD"
                }
            },
            "cost": {
                "raw": {
                    "value": 7108.8,
                    "units": "USD"
                },
                "markup": {
                    "value": 1421.76,
                    "units": "USD"
                },
                "usage": {
                    "value": 0.0,
                    "units": "USD"
                },
                "total": {
                    "value": 8530.56,
                    "units": "USD"
                }
            }
        }
    },
```
